### PR TITLE
Add and remove proxies on all proxy method args

### PR DESCRIPTION
**Pull request by milliams**
Until now it seems that while schema attributes were being correctly stripped and wrapped as they were passed through the proxy-interface, the same was not being done for arguments and return values of exported methods. 

All proxy-exported methods are accessed via `ProxyMethodDescriptor` which now correctly removes the proxies from all incoming arguments and adds them to any return value.

This depends on #392 to make one of the `GangaList` tests pass so I will rebase this on develop once that is merged.

### CODE REVIEW

File: python/Ganga/GPIDev/Base/Proxy.py [hunk 1]
```diff
@@ -626,6 +627,24 @@ def __set__(self, obj, _val):
...
+    # type: (Callable) -> Callable
+    """
+    A decorator to strip the proxy from all incoming arguments
+    (including ``self`` if it's a method) and add one to the return
+    value.
+    """
+
+    @functools.wraps(f)
+    def proxy_wrapped(*args, **kwargs):
+        s_args = [stripProxy(a) for a in args]
```
Comment (rob-c) [inline python/Ganga/GPIDev/Base/Proxy.py]: Is this recursive?

Comment (milliams) [inline python/Ganga/GPIDev/Base/Proxy.py]: No, and it needn't be since only the outer-most object in a hierarchy is proxy-wrapped.

Comment (alexanderrichards) [inline python/Ganga/GPIDev/Base/Proxy.py]: indeed, you beat me to it @milliams. This is the beauty (IMO) of the system that I am using. The proxy is just an outermost layer that object cross. e.g imagine a list of objects, as you fill that list the objects are unwrapped and stored in the list without proxies. Their proxies are applied again when you get them out.

Comment (rob-c) [inline python/Ganga/GPIDev/Base/Proxy.py]: As long as we're now stripping the proxies everywhere then OK I'm convinced, I'm just weary that users are usually good at finding corner cases such as the proxies which don't use the automatic format.

**Comment by rob-c** [comment]
Well if the tests are passing then go for me, would be nice to see an auto-proxy wrapper to add proxies to objects returned automatically.

File: python/Ganga/GPIDev/Base/Proxy.py (hunk 1 above)
Comment (milliams) [inline python/Ganga/GPIDev/Base/Proxy.py]: Indeed, things with non-automatic proxies (like the registries etc.) are a potential leak in the proxy layer but they should be fixed by making them conform to the `_impl`/`_proxyObject` interface so that `assert a is addProxy(stripProxy(a))` always works.

The registries aren't usually passed through the proxy layer though so they're unlikely to be a problem in the short-term.

**Comment by milliams** [comment]
What do you mean by an "auto-proxy wrapper"? `proxy_wrap` already wraps the return value of functions.

**Comment by rob-c** [comment]
In that case I must have been looking at the wrong piece of code, I'm still happy to merge
